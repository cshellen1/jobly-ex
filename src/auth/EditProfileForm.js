import React, { useContext, useState } from "react";
import JoblyApi from "../api/api";
import UserContext from "./UserContext";

function EditProfileForm() {
	const { currentUser, setCurrentUser } = useContext(UserContext);

	const [formData, setFormData] = useState({
		email: currentUser.email,
		password: "",
		lastName: currentUser.lastName,
    firstName: currentUser.firstName,
    username: currentUser.username,
	});

  async function updateUser(data) {
    try {
      const updatedUser = await JoblyApi.updateUser(currentUser.username, data);
      setCurrentUser(updatedUser);
    } catch (error) {
      console.log(error);
    }
	}

	function handleChange(e) {
		const { name, value } = e.target;
		setFormData((formData) => ({ ...formData, [name]: value }));
	}

	function handleSubmit(e) {
    e.preventDefault();
    let updateData = {
      email: formData.email,
      password: formData.password,
      lastName: formData.lastName,
      firstName: formData.firstName,
    }
    try{
      updateUser(updateData);
    }
    catch(error){
      console.log(error);
    }
    setFormData((f) => ({ ...f, password: "" }));
    alert("Profile Updated");
	}

	return (
		<form className="col-md-6" onSubmit={handleSubmit}>
			<div className="form-group">
				<label>Username</label>
				<p className="form-control-plaintext">{formData.username}</p>
				<label htmlFor="firstName">First Name</label>
				<input
					type="text"
					className="form-control w-25"
					id="firstName"
					name="firstName"
					value={formData.firstName}
					onChange={handleChange}
				/>
			</div>
			<div className="form-group">
				<label htmlFor="lastName">Last Name</label>
				<input
					type="text"
					className="form-control w-25"
					id="lastName"
					name="lastName"
					value={formData.lastName}
					onChange={handleChange}
				/>
			</div>
			<div className="form-group">
				<label htmlFor="email">Email</label>
				<input
					type="text"
					className="form-control w-25"
					id="email"
					name="email"
					value={formData.email}
					onChange={handleChange}
				/>
			</div>
			<div className="form-group">
				<label htmlFor="password">Confirm password to make changes</label>
				<input
					type="password"
					className="form-control w-25"
					id="password"
					name="password"
					value={formData.password}
					onChange={handleChange}
				/>
			</div>
			<button type="submit" className="btn btn-primary">
				Edit
			</button>
		</form>
	);
}

export default EditProfileForm;
