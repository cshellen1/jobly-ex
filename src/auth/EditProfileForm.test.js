import React from "react";
import { render } from "@testing-library/react";
import EditProfileForm from "./EditProfileForm";
import { UserProvider } from "../testUtils";
import { MemoryRouter } from "react-router";

// TODO: woefully under-tested!

it("renders without crashing", function () {
	render(
		<MemoryRouter>
			<UserProvider>
				<EditProfileForm />
			</UserProvider>
		</MemoryRouter>
	);
});

it("matches snapshot", function () {
	const { asFragment } = render(
		<UserProvider>
			<EditProfileForm />
		</UserProvider>
	);
	expect(asFragment()).toMatchSnapshot();
});

it("displays the correct values", function () {
	const { getByLabelText } = render(
		<UserProvider>
			<EditProfileForm />
		</UserProvider>
	);
	expect(getByLabelText("First Name")).toHaveValue("testfirst");
	expect(getByLabelText("Last Name")).toHaveValue("testlast");
	expect(getByLabelText("Email")).toHaveValue("test@test.net");
});
