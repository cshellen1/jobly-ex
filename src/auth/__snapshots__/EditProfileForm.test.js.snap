// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="col-md-6"
  >
    <div
      class="form-group"
    >
      <label>
        Username
      </label>
      <p
        class="form-control-plaintext"
      >
        testuser
      </p>
      <label
        for="firstName"
      >
        First Name
      </label>
      <input
        class="form-control w-25"
        id="firstName"
        name="firstName"
        type="text"
        value="testfirst"
      />
    </div>
    <div
      class="form-group"
    >
      <label
        for="lastName"
      >
        Last Name
      </label>
      <input
        class="form-control w-25"
        id="lastName"
        name="lastName"
        type="text"
        value="testlast"
      />
    </div>
    <div
      class="form-group"
    >
      <label
        for="email"
      >
        Email
      </label>
      <input
        class="form-control w-25"
        id="email"
        name="email"
        type="text"
        value="test@test.net"
      />
    </div>
    <div
      class="form-group"
    >
      <label
        for="password"
      >
        Confirm password to make changes
      </label>
      <input
        class="form-control w-25"
        id="password"
        name="password"
        type="password"
        value=""
      />
    </div>
    <button
      class="btn btn-primary"
      type="submit"
    >
      Edit
    </button>
  </form>
</DocumentFragment>
`;
